aws/fsx.go

```diff
@@ -48,6 +48,26 @@ func refreshFsxFileSystemLifecycle(conn *fsx.FSx, id string) resource.StateRefre
 	}
 }
 
+func refreshFsxFileSystemLifecycleOptimizing(conn *fsx.FSx, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		filesystem, err := describeFsxFileSystem(conn, id)
+
+		if isAWSErr(err, fsx.ErrCodeFileSystemNotFound, "") {
+			return nil, "", nil
+		}
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		if filesystem == nil {
+			return nil, "", nil
+		}
+
+		return filesystem, aws.StringValue(filesystem.AdministrativeActions[0].Status), nil
```

Comment:
This line is missing a length check on `AdministrativeActions`, which can cause a panic. Created followup issue: https://github.com/hashicorp/terraform-provider-aws/issues/16440